Sources/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegate.swift

```diff
@@ -29,16 +29,18 @@ open class DDURLSessionDelegate: NSObject, URLSessionTaskDelegate, URLSessionDat
     }
 
     var instrumentation: URLSessionAutoInstrumentation? {
-        core().v1.feature(URLSessionAutoInstrumentation.self)
+        core?.v1.feature(URLSessionAutoInstrumentation.self)
     }
 
     let firstPartyHosts: FirstPartyHosts
 
-    private let core: () -> DatadogCoreProtocol
+    /// The instance of the SDK core notified by this delegate.
+    /// It must be a weak reference, because `URLSessionDelegate` can last longer than this instance.
+    private weak var core: DatadogCoreProtocol?
```

Comment:
Creating could happen before initialising the SDK, that's why we were using a closure here, to lazily get the core.

Reply:
Good catch @maxep - I made a change (pls 🙏 review as it introduces some new concepts). Also added unit tests for all initializers, so we don't miss it again.